### PR TITLE
feat(blocks): support connections between groups

### PR DESCRIPTION
**Pull request by fundon**
Closes: [BS-431](https://linear.app/affine-design/issue/BS-431/group-connect-group), [AFF-1184](https://linear.app/affine-design/issue/AFF-1184/connector-可以连接-group-和-group-内的yuan素)

### What's changed?

* Allow groups connect other groups 
* Allow groups connect elements within other groups
* Allow elements within groups connect other groups
* Highlight groups

### CODE REVIEW

**Comment by fundon** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/7192?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#7192** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/7192?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#7167** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/7167?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @fundon and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by Svaney-ssman — COMMENTED** [review]
According to [design](https://www.figma.com/design/i4Za2oTwsiEPqbRpny1Q1M/%F0%9F%8F%98%EF%B8%8F-2024-S2?node-id=4623-17315&t=m791EG6SQq2x0LK0-4)
Active any one of the groups will also active the connector and the group indicators

**Comment by Saul-Mirone** [comment]
@doouding PTAL

File: packages/blocks/src/surface-block/managers/connector-manager.ts [hunk 1]
```diff
@@ -891,29 +922,42 @@ export class ConnectionOverlay extends Overlay {
...
         )
       ) {
+        target = connectable;
         result = {
           id: connectable.id,
         };
       }
     }
 
+    if (target && target instanceof GroupElementModel) {
```
Comment (doouding) [inline packages/blocks/src/surface-block/managers/connector-manager.ts:932]: I think there's no need to check nullish status.

**Review by doouding — APPROVED** [review]

**Review by Saul-Mirone — APPROVED** [review]

**Pull request merged**